nucypher/policy/models.py

```diff
@@ -534,140 +534,166 @@ def __len__(self):
+            capsule, signature, remainder = item_splitter(data, return_remainder=True)
+            if remainder:
+                remainder_splitter = BytestringSplitter((CapsuleFrag, VariableLengthBytestring), Signature)
+                cfrag, reencryption_signature = remainder_splitter(remainder)
+                return cls(capsule=capsule, signature=signature,
+                           cfrag=cfrag, reencryption_signature=reencryption_signature)
+            else:
+                return cls(capsule=capsule, signature=signature)
+
+        def attach_work_result(self, cfrag, reencryption_signature):
+            self.cfrag = cfrag
+            self.reencryption_signature = reencryption_signature
 
     def __init__(self,
                  bob: Bob,
                  arrangement_id,
-                 capsules: List[Capsule],
-                 capsule_signatures: List[Signature],
                  alice_address: bytes,
-                 alice_address_signature: bytes,
-                 receipt_bytes: bytes,
+                 tasks: List,
                  receipt_signature,
                  ursula=None,
+                 blockhash=None
                  ) -> None:
         self.bob = bob
         self.arrangement_id = arrangement_id
-        self.capsules = capsules
-        self.capsule_signatures = capsule_signatures
         self.alice_address = alice_address
-        self.alice_address_signature = alice_address_signature
-        self.receipt_bytes = receipt_bytes
+        self.tasks = tasks
         self.receipt_signature = receipt_signature
         self.ursula = ursula  # TODO: We may still need a more elegant system for ID'ing Ursula.  See #136.
+        self.blockhash = blockhash or b'\x00' * 32  # TODO
         self.completed = False
 
     def __repr__(self):
         return "WorkOrder for hrac {hrac}: (capsules: {capsule_bytes}) for Ursula: {node}".format(
             hrac=self.arrangement_id.hex()[:6],
-            capsule_bytes=[binascii.hexlify(bytes(cap))[:6] for cap in self.capsules],
+            capsule_bytes=[binascii.hexlify(bytes(item.capsule))[:6] for item in self.tasks],
             node=binascii.hexlify(bytes(self.ursula.stamp))[:6])
 
     def __eq__(self, other):
-        return (self.receipt_bytes, self.receipt_signature) == (
-            other.receipt_bytes, other.receipt_signature)
+        return self.receipt_signature == other.receipt_signature
 
     def __len__(self):
-        return len(self.capsules)
+        return len(self.tasks)
 
     @classmethod
     def construct_by_bob(cls, arrangement_id, capsules, ursula, bob):
         alice_verifying_key = capsules[0].get_correctness_keys()["verifying"]
+        alice_address = canonical_address_from_umbral_key(alice_verifying_key)
+
+        # TODO: Bob's input to prove freshness for this work order
+        blockhash = b'\x00' * 32
 
-        capsules_bytes = []
-        capsule_signatures = []
+        tasks, tasks_bytes = [], []
```

Comment:
Am I the only one who enjoys reading `list()`?  :smile:

Reply:
I think so. I like `[]` more!